lib/msal-common/src/authority/OIDCOptions.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Options for the OIDC protocol mode.
+ */
+export type OIDCOptions = {
+    serverResponseType?: ServerResponseType;
+    defaultScopes?: Array<string>;
+};
+
+export const ServerResponseType = {
+    QUERY: "query",
+    HASH: "hash",
```

Comment:
Can we reuse the `ResponseMode` enum here instead of creating a new one?

Reply:
I made a new one because ResponseMode also has form_post, which we don't want for OIDCOptions.

Reply:
You can create a subtype called OidcResponseMode to re-use the ResponseMode type like so:

```typescript
export const OidcResponseMode = Exclude<ResponseMode, ResponseMode.FORM_POST>
```

Removes as much repetition as possible. For readability I'd put that type in Constants right after ResponseMode and import it wherever it's used.